File: core/src/main/java/core/qna/QnaSystem.java
package core.qna;

import core.rag.OpinionDTO;
import core.rag.QuestionDTO;
import core.rag.QuestionTitleDTO;

public interface QnaSystem {
    String enrollQuestion(String userId, String title, String category, String contents);
    String updateQuestion(String userId, String questionId, String category, String contents);
    String deleteQuestion(String userId, String questionId);
    QuestionDTO readQuestion(String questionId);
    QuestionTitleDTO[] readQuestionTitles(String category, int startNum, int endNum);

    String enrollOpinion(String userId, String questionId, String title, String contents);
    String updateOpinion(String userId, String opinionId, String title, String contents);
    String deleteOpinion(String userId, String opinionId);
    OpinionDTO[] readOpinions(String questionId, int startNum, int endNum);
}

File: core/src/main/kotlin/core/qna/QnaSystem.kt
package core.qna

import core.rag.Opinion
import core.rag.Question
import core.rag.QuestionTitle
import core.rag.event.QnAEvent

interface QnaSystem {
    fun enrollQuestion(event: QnAEvent.EnrollQuestion): Question
    fun updateQuestion(event: QnAEvent.UpdateQuestion)
    fun deleteQuestion(event: QnAEvent.DeleteQuestion)
    fun readQuestion(questionId: String): Question
    fun readQuestionTitles(category: String, offset: Int, limit: Int): List<QuestionTitle>

    fun enrollOpinion(event: QnAEvent.EnrollOpinion)
    fun updateOpinion(event: QnAEvent.UpdateOpinion)
    fun deleteOpinion(event: QnAEvent.DeleteOpinion)
    fun readOpinions(questionId: String, offset: Int, limit: Int): List<Opinion>
}